ec2/spark_ec2.py

```diff
@@ -37,12 +37,17 @@
 import tempfile
 import textwrap
 import time
-import urllib2
 import warnings
 from datetime import datetime
 from optparse import OptionParser
 from sys import stderr
 
+try:
```

Comment:
Instead of using try/except to handle cross-version compatibility, how about explicitly checking `sys.version` here, like we do in a few other files?  This isn't a huge deal, but I think it's clearer to explicitly check versions when imports differ due to cross-version compatibility reasons (it's fine to use try/except for things like `cStringIO`, where we're trying to fall back if native modules aren't available).